core/src/main/scala/com/mesosphere/usi/core/Scheduler.scala

```diff
@@ -130,4 +149,25 @@ object Scheduler {
       }
     }
   }
+
+  private def persistenceFlow(
+      podRecordRepository: PodRecordRepository): Flow[SchedulerEvents, SchedulerEvents, NotUsed] =
+    Flow[SchedulerEvents].mapAsync(1) { events =>
+      import CallerThreadExecutionContext.context
+      Future
+        .sequence(events.stateEvents.map {
```

Comment:
I would go for something like:
```scala
mapAsync(1) { events => 
val (updated, deleted) = events.stateEvents.partition(....)
val updatedF: Future[Done] = Source(updated)
 .via(podRepository.storeFlow)
 .runWith(Sink.ignore)

val deletedF: Future[Done] = Source(deleted)
 .via(podRepository.deleteFlow)
 .runWith(Sink.ignore)

updated.andThen(deleted) 
}
```
It has the downside of 2x materialization but that might be ok for now.